theano/gpuarray/opt.py

```diff
@@ -393,53 +395,35 @@ def apply(self, fgraph):
             new_nodes.append(new_o)
         fgraph.replace_all_validate(zip(fgraph.outputs, new_nodes))
 
+        end_nb_nodes = len(fgraph.apply_nodes)
+
+        return (self, start_nb_nodes, end_nb_nodes, max_nb_nodes, io_toposort_timing,
+                nb_nodes, time_opts, node_created)
+
     @staticmethod
     def print_profile(stream, prof, level=0):
-        (opt, loop_timing, loop_process_count,
-         (start_nb_nodes, end_nb_nodes, max_nb_nodes),
-         local_opt_timing, nb_nodes, time_opts, io_toposort_timing,
-         node_created) = prof
+        (opt, start_nb_nodes, end_nb_nodes, max_nb_nodes, io_toposort_timing,
+         nb_nodes, time_opts, node_created) = prof
         blanc = ('    ' * level)
         print(blanc, "GraphToGPUOptimizer", end=' ', file=stream)
         print(blanc, getattr(opt, "name",
                              getattr(opt, "__name__", "")), file=stream)
-        print(blanc, "  time %.3fs for %d passes" % (
-            sum(loop_timing), len(loop_timing)), file=stream)
         print(blanc, "  nb nodes (start, end,  max) %d %d %d" % (
             start_nb_nodes, end_nb_nodes, max_nb_nodes), file=stream)
         print(blanc, "  time io_toposort %.3fs" % sum(
             io_toposort_timing), file=stream)
-        s = sum([time_opts[o] for o in opt.local_optimizers_all])
-        print(blanc, "  time in local optimizers %.3fs" % s, file=stream)
+
+        s = sum([time_opts[o] for o in opt.new_opts])
         
-        for i in range(len(loop_timing)):
-            lopt = ""
-            if loop_process_count[i]:
-                d = list(reversed(sorted(iteritems(loop_process_count[i]),
-                                         key=lambda a: a[1])))
-                lopt = " ".join([str((str(k), v)) for k, v
-                                 in d[:5]])
-                if len(d) > 5:
-                    lopt += " ..."
-            print(blanc, ('  %2d - %.3fs %d (%.3fs in global opts, '
-                          '%.3fs io_toposort) - %d nodes - %s' % (
-                              i, loop_timing[i],
-                              sum(loop_process_count[i].values()),
-                              local_opt_timing[i],
-                              io_toposort_timing[i], nb_nodes[i],
-                              lopt)), file=stream)
+        print(blanc, "  time in local optimizers %.3fs" % s, file=stream)
 
         count_opt = []
         not_used = []
         not_used_time = 0
         process_count = {}
-        for o in (opt.local_optimizers_all +
-                  list(opt.local_optimizers_map.get(type(node.op), [])) +
-                  list(opt.local_optimizers_map.get(node.op, []))):
+        for o in (opt.new_opts):
             process_count.setdefault(o, 0)
-        for count in loop_process_count:
-            for o, v in iteritems(count):
-                process_count[o] += v
```

Comment:
removing those loops will cause process_count to be always 0. I think in the new code, you must do:

for o in opt.new_opts:
   process_count.setdefault(o, 0)
   process_count[o] + 1